gpytorch/lazy/kronecker_product_lazy_tensor.py

```diff
@@ -151,11 +151,26 @@ def add_diag(self, diag):
     @cached(name="symeig")
     def _symeig(self, eigenvectors=True):
         # eigenvectors may not get zeroed if called w/o eigenvectors after initialization
+        from torch import zeros_like
 
         evals, evecs = [], []
         for lazy_tensor in self.lazy_tensors:
             # TODO: replace with lazy_tensor.symeig() once that is added in.
-            evals_, evecs_ = lazy_tensor.evaluate().symeig(eigenvectors=eigenvectors)
+            # TODO: ensure that the symeig call is also done in this manner
+
+            eval_tensor = lazy_tensor.evaluate()
+            tensor_dtype = eval_tensor.dtype
+
+            evals_, evecs_ = eval_tensor.double().symeig(eigenvectors=eigenvectors)
+
+            # we chop any negative eigenvalues
+            neg_evals = zeros_like(evals_)
+            neg_evals[evals_.data < 0] = -evals_.data[evals_.data < 0]
```

Comment:
nit: assign `evals_.data < 0` to local var to avoid doing the comparison twice

Reply:
Are these lines just `evals_.clamp_min_(0)`?

Reply:
Somehow, I didn't realize that existed.